EventGenerator/src/CaloCalibGun_module.cc

```diff
@@ -0,0 +1,241 @@
+  CaloCalibGun::CaloCalibGun(const Parameters& conf)
+    : EDProducer{conf}
+    , _mean{conf().mean()}
+    , _energy{conf().energy()}
+    , _cosmin{conf().cosmin()}
+    , _cosmax{conf().cosmax()}
+    , _phimin{conf().phimin()}
+    , _phimax{conf().phimax()}
+    , _tmin{conf().tmin()}
+    , _tmax{conf().tmax()}
+    , _nDisk{conf().nDisk()}
+    , _engine{createEngine(art::ServiceHandle<SeedService>()->getSeed())}
+    , _randFlat{_engine}
+    , _randPoissonQ{_engine, _mean*1.0}
+    , _randomUnitSphere{_engine, _cosmin, _cosmax, 0, CLHEP::twopi}
+  {
+    produces<mu2e::GenParticleCollection>();
+    produces<mu2e::PrimaryParticle>();
+    produces <MCTrajectoryCollection>();
+
+    if (_mean < 0) throw cet::exception("RANGE") << "CaloCalibGun.mean must be non-negative "<< '\n';
+  }
+
+  //================================================================
+  void CaloCalibGun::produce(art::Event& event) {
+    std::unique_ptr<GenParticleCollection> output(new GenParticleCollection);
+    PrimaryParticle primaryParticles;
+    MCTrajectoryCollection mctc;
+
+    const DiskCalorimeter *_cal = GeomHandle<DiskCalorimeter>().get();
+    //_nPipes          = _cal->caloInfo().getInt("nPipes");
+    _pipeRadius      = _cal->caloInfo().getDouble("pipeRadius");
+    _pipeTorRadius   = _cal->caloInfo().getVDouble("pipeTorRadius");
+    _randomRad       = _cal->caloInfo().getVDouble("pipeTorRadius");
+    _zPipeCenter     = _cal->disk(_nDisk).geomInfo().origin()-CLHEP::Hep3Vector(0,0,_cal->disk(_nDisk).geomInfo().size().z()/2.0-_pipeRadius);//disk =1
+    // we normalize to the volume of the pipe (proportional to 2*pi*R if they have all the same radius) to draw a random number from which to generate the photons TODO - is this used?
+    double sumR(0);
+    std::for_each(_randomRad.begin(), _randomRad.end(), [&](double& d) {sumR+=d; d = sumR; });
+    std::for_each(_randomRad.begin(), _randomRad.end(), [&](double& d) {d /= sumR;});
+ 
+    //Define the parameters of the pipes:
+    
+    // angle of large torus in degrees
+    std::vector<double> phi_lbd = _cal->caloInfo().getVDouble("largeTorPhi");
+    // angle of small torus in degrees
+    std::vector<double> phi_sbd = _cal->caloInfo().getVDouble("smallTorPhi");
+    // angle of the end point
+    std::vector<double> phi_end = _cal->caloInfo().getVDouble("straightEndPhi");
+    // center position y of the small torus
+    std::vector<double> ysmall = _cal->caloInfo().getVDouble("yposition");
+    // radius of small torus
+    double radSmTor = _cal->caloInfo().getDouble("radSmTor");
+    // first center position x of the small torus
+    double xsmall = _cal->caloInfo().getDouble("radSmTor");
+    // distance of the small torus center
+    double xdistance = _cal->caloInfo().getDouble("xdistance");
+    // inner radius of the manifold
+    double rInnerManifold = _cal->caloInfo().getDouble("rInnerManifold");
+    vector<float> sign{-1.0, 1.0};
+
+    for (unsigned int ig = 0; ig < _mean; ++ig) 
```

Comment:
You create a Poisson random engine but never use it.  If your intention is to make the number of photons per event Poisson distributed, then use it.  If not, then remove it; if you do remove it,  then the name of the variable _mean is misleading so change it to to something that correctly describes it's use, such as nPhotons.  Note that they type of _mean is unsigned which suggests it was never intended to be a Poisson mean, which can take on values within the domain of Real ( in the mathematical sense).